sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,375 @@
+ /*  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));  */     
+}
+
+/**
+ * External ISR.
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this OTTF ISR. This ISR
+ * overrides the default OTTF implementation.
+ */
+void ottf_external_isr(void) {
+  top_earlgrey_plic_peripheral_t peripheral_serviced;
+  dif_alert_handler_irq_t irq_serviced;
+  isr_testutils_alert_handler_isr(plic_ctx, alert_handler_ctx,
+                                  &peripheral_serviced, &irq_serviced);
+  CHECK(peripheral_serviced == kTopEarlgreyPlicPeripheralAlertHandler,
+        "Interurpt from unexpected peripheral: %d", peripheral_serviced);
+}
+
+bool test_main(void) {
+  init_peripherals();
+
+  //alert_handler_config();
+
+  // Enable the external IRQ at Ibex.
+  //irq_global_ctrl(true);
+  //irq_external_ctrl(true);
+
+  int aa;
+  bool is_sth;
+
+//---------------------- INITIAL ALERT_HANDLER CONFIG ---------------
+  LOG_INFO("STEP #1: ALERT_HANDLER BEFORE THE CONFIG");
+  // Print if any of the alerts is locked
+  // Expected behavior: no print
+  for(aa=0; aa<63; aa++) {
+    CHECK_DIF_OK(dif_alert_handler_is_alert_locked( &alert_handler, aa, &is_sth));
+    if (is_sth)
+    {
+      LOG_INFO("is_locked_alert_%d = %d", aa, is_sth);
+    }
+  }
+  
+  // Print if any of the local_alerts is locked
+  // Expected behavior: no print
+  for(aa=0; aa<7; aa++) {
+    CHECK_DIF_OK(dif_alert_handler_is_local_alert_locked( &alert_handler, aa, &is_sth));
+    if (is_sth)
+    {
+      LOG_INFO("is_locked_local_alert_%d = %d", aa, is_sth);
+    }
+  }
+
+  // Print if any of the alert classess is locked
+  // Expected behavior: no print
+  for(aa=0; aa<4; aa++) {
+    CHECK_DIF_OK(dif_alert_handler_is_class_locked( &alert_handler, aa, &is_sth));
+    if (is_sth)
+    {
+      LOG_INFO("is_locked_class_%d = %d", aa, is_sth);
+    }
+  }
+
+  // Print if ping timer is locked
+  // Expected behavior: no print
+  CHECK_DIF_OK(dif_alert_handler_is_ping_timer_locked( &alert_handler, &is_sth));
+  if (is_sth)
+  {
+    LOG_INFO("is_locked_ping_timer", is_sth);
+  }
+  LOG_INFO("\n");
+  //-----------------------------------------------------------------------
+  
+  dif_toggle_t clock_state;
+
+// TRANSACTIONAL CLOCK ENABLE/DISABLE
+  
+  // Enable/disable AES clocks => alert #42 (recov), alert #43(fatal) (somehow worked)
+/* 
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kTopEarlgreyHintableClocksMainAes,kDifToggleDisabled));
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_get_enabled(&clkmgr, kTopEarlgreyHintableClocksMainAes, &clock_state));
+  LOG_INFO("Clock enabled state hintable (%d).", clock_state); 
+*/
+
+// Enable/disable HMAC clocks => alert #43 (did not work)
```

Comment:
I wonder in your code, did you use `43`? I think HMAC should be alert `44`?

Reply:
It seems the comments are outdated, sorry for the misleading comments. I used the correct indices during my tests.I tested the alerts from 42 to 48 individually as well as together.